components/script/dom/htmlelement.rs

```diff
@@ -151,7 +151,12 @@ impl HTMLElementMethods for HTMLElement {
     // https://html.spec.whatwg.org/multipage/#handler-onload
     fn GetOnload(&self) -> Option<Rc<EventHandlerNonNull>> {
         if self.is_body_or_frameset() {
-            window_from_node(self).GetOnload()
+            let document = document_from_node(self);
+            if document.has_browsing_context() {
+                document.window().GetOnload()
+            } else {
+                None
```

Comment:
That's orthogonal to the problem I described. If there isn't a browsing context (i.e. there's not a window), then you should check whether the HTML element has an event handler registered for this particular event.